.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,205 @@
+
+* `gh-pages`: Is only touched by the administrator and for maintaining the [landing page](http://sysbiochalmers.github.io/yeast-GEM/) of yeast-GEM.
+
+* `{chore, doc, feat, fix, refactor, style}/descriptive-name`: Any other branch created in the model. If you work on a fix, start the branch name with `fix/`, if you work on a feature, start the branch name with `feat/`. Examples: `fix/format_reactions` or `feat/new_algorithms`. [See below](#semantic-commits) for more details on the possible actions you can use.
+	
+#### Semantic commits
+
+Please use concise descriptive commit messages. Ideally, use semantic commit messages to make it easier to show what you are aiming to do:
+
+`action-object: brief description`
+* `action` = {`feat`, `fix`, `refactor`, `style`, `doc`, `chore`, `test`}
+* `object` = {`rxn`, `rxn.annot`, `rxn.prop`, `met`, `met.annot`, `met.prop`, `gene`, `gene.annot`, `comp`, `comp.annot`, `data`}
+
+`action` refers to what exactly are you doing in the commit, following a [standard definition](http://karma-runner.github.io/2.0/dev/git-commit-msg.html) in software development: 
+* `chore`: updating toolbox, data files, etc.
+* `doc`: updating documentation or explanatory comments in functions.
+* `feat`: new feature added, e.g. new reaction / metabolite / function / etc.
+* `fix`: something that was incorrect in the model and now has been corrected.
+* `refactor`: see [code refactoring](https://en.wikipedia.org/wiki/Code_refactoring).
+* `style`: minor format changes of model, functions or data (spaces, semi-colons, etc., no code change).
+
+`object` may or may not be used, and refers to the receiver of the action, i.e. what part of the model/data are you modifying:
+* `object`: Used when adding/removing/modifying the whole object (`rxn`, `met`, `gene`, `comp` or `data`).
+* `object.annot`: Annotation of the object (e.g. KEGG IDs, EC numbers, etc). Applies for `rxn`, `met`, `gene` or `comp`.
+* `object.prop`: Any other property of the object (e.g. formulas, charge, stoich coeffs, LB, UB, etc). Applies for `rxn` or `met`.
+
+Some examples:
+
+|commit|commit message|
+|:---:|:---:|
+|Add new rxns|`feat-rxn: methanol pathway`|
+|Remove a metabolite|`fix-met: duplicated citrate`|
+|Add metabolite formula|`feat-met.prop: carbohydrate formulas`|
+|Fix rxn stoichiometry|`fix-rxn.prop: complex V stoich coeffs`|
+|Update gene IDs|`fix-gene.annot: update IDs from SWISSPROT`|
+|Format name of compartment|`style-comp.annot: remove uppercases`|
+|Split a rxn in 2|`refactor-rxn: split isomerase in 2 steps`|
+|Add some data|`feat-data: metabolomics data`|
+|Update documentation of function|`doc: addDBnewRxn.m`|
+|Update toolbox|`chore: update RAVEN version`|
+
+More examples [here](https://github.com/SysBioChalmers/yeast-GEM/commits/master). A more detailed explanation or comments can be left in the commit description.
```

Comment:
Shouldn't more detailed explanations be encouraged?

Reply:
done